bindings/gumjs/gumscript-runtime-dalvik.js

```diff
@@ -258,6 +265,67 @@
             return new C(C.__handle__, null);
         };
 
+        this.choose = function (className, callbacks) {
+            let env = vm.getEnv();
+            let klass = this.use(className);
+
+            let enumerateInstances = function (className, callbacks) {
+                let thread = Memory.readPointer(env.handle.add(JNI_ENV_OFFSET_SELF));
+                let ptrClassObject = api.dvmDecodeIndirectRef(thread, klass.$classHandle);
+
+                let pattern = ptrClassObject.toMatchPattern();
+                let heapSourceBase = api.dvmHeapSourceGetBase();
+                let heapSourceLimit = api.dvmHeapSourceGetLimit();
+
+                let size = heapSourceLimit.toInt32() - heapSourceBase.toInt32();
+                Memory.scan(heapSourceBase, size, pattern, {
+                    onMatch: function (address, size) {
+                        if (api.dvmIsValidObject(address)) {
+                            Dalvik.perform(function () {
+                                let env = vm.getEnv();
+                                let thread = Memory.readPointer(env.handle.add(JNI_ENV_OFFSET_SELF));
+                                let localReference = api.addLocalReferenceFunc(thread, address);
+                                let instance = Dalvik.cast(localReference, klass);
+                                let stopMaybe = callbacks.onMatch(instance);
```

Comment:
It would be good to use `const` instead of `let` for variables that are only assigned to once in the current scope.